app/controllers/sys_controller.rb

```diff
@@ -99,14 +101,37 @@ def check_enabled
   private
 
   def require_basic_auth
+    @authenticated_user =
+    if Setting.repository_authentication_bypass?
+      find_user_from_basic_auth
+    else
+      # Authenticate the user internally
+      perform_basic_auth
+    end
+
+    render_authorization_required unless @authenticated_user.present?
```

Comment:
I think this kind of assignment is confusing too. But I'm no expert on this matter, nor does my opinion have any special weight.